File: src/programmers_prac/lessons12915.java
//연습문제 > 문자열 내 마음대로 정렬하기
package programmers_prac;

import java.util.Arrays;
import java.util.Comparator;

public class lessons12915 {
	public static String[] solution(String[] strings, int n) {
		String[] answer = {};

		Arrays.sort(strings, new Comparator<String>() {
			@Override
			public int compare(String o1, String o2) {
				if (o1.charAt(n) > o2.charAt(n)) {
					return 1;
				}
				if (o1.charAt(n) < o2.charAt(n)) {
					return -1;
				}
				if (o1.charAt(n) == o2.charAt(n)) {
					if (o1.compareTo(o2) >= 1) {
						return 1;
					} else {
						return -1;
					}
				}
				return 0;
			}
		});
		for (String a : strings) {
			System.out.println(a);
		}
		return strings;
	}

	public static void main(String[] args) {
		String[] a = { "aba", "abad", "abc", "acca", "abca" };
		solution(a, 2);
	}

}

File: src/programmers_prac/lessons12915.js
function solution(strings, n) {
    var answer = [];

    strings.sort((o1, o2)=>{
        if(o1[n] === o2[n]){
            if(o1 > o2){
                return 1;
            }else if(o1 < o2){
                return -1;
            }else{
                return 0;
            }
        }else{
            if(o1[n] > o2[n]){
                return 1;
            }
            else{
                return -1;
            }
        }
    })
    return strings;
}
let a = "avs";
let b = "avs"

console.log(
    a.localeCompare(b));